core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -513,13 +516,44 @@ private[spark] object Utils extends Logging {
       Files.move(sourceFile, destFile)
```

Comment:
Hmmm...

    scala> val from = new File("/tmp/foo")
    from: java.io.File = /tmp/foo
    
    scala> val to = new File("/home/vanzin/tmp/foo")
    to: java.io.File = /home/vanzin/tmp/foo
    
    scala> Files.move(from, to)
    java.io.FileNotFoundException: /tmp/foo (Is a directory)
            at java.io.FileInputStream.open(Native Method)
            at java.io.FileInputStream.<init>(FileInputStream.java:120)
            at com.google.common.io.Files$FileByteSource.openStream(Files.java:124)
            at com.google.common.io.Files$FileByteSource.openStream(Files.java:114)
            at com.google.common.io.ByteSource.copyTo(ByteSource.java:202)
            at com.google.common.io.Files.copy(Files.java:436)
            at com.google.common.io.Files.move(Files.java:651)

I think it doesn't work if `from` and `to` are on different file systems (such as in my case above).